tests/replacer/test_single_nodes.py

```diff
@@ -0,0 +1,342 @@
+from collections import defaultdict
```

Comment:
Could we rename this file to `test_replacer.py` to mirror the file being tested? I didn't get what the "single nodes" refers to.